### PR TITLE
Explosion blockers system and bomb to loot machine for toxins.

### PR DESCRIPTION
**Pull request by AnturK**
Simple explosion blocker system + machine that turns bombs into "fusion cells" which are mostly placeholder for now.

What's this about : 
Certain things can now affect or react to explosions inside them.
Toxin machine turns whatever is exploding inside into a cell with devastation range rating. (For now it only sells in cargo)

TODO:
- [ ] Better name & fluff
- [ ] Sprites
- [ ] Some actual use besides cargo

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets/utility_closets.dm [hunk 1]
```diff
@@ -153,6 +153,18 @@
 	new /obj/item/clothing/shoes/sneakers/black( src )
 	new /obj/item/clothing/head/bomb_hood( src )
 
+//It's bomb-resistant closet get it
+/obj/structure/closet/bombcloset/handle_explosion(atom/source, devastation_range, heavy_impact_range, light_impact_range, flash_range, adminlog = TRUE, ignorecap = FALSE, flame_range = 0 , silent = FALSE, smoke = FALSE)
```
Comment (AnturK) [inline code/game/objects/structures/crates_lockers/closets/utility_closets.dm]: This is mostly debug, i'll remove it later

**Comment by optimumtact** [comment]
Wow you made a shitty point system as well? Stop making these, they're not GOOD gameplay loops

**Comment by AnturK** [comment]
What point system ? These are just here to turn bombs into something more productive.

**Comment by optimumtact** [comment]
You're turning bombs into a virtual equivalent of points, it's not a good game mechanic, solving the puzzle of getting a bomb should be the reward, not some arbitrary points system tacked on at the end.

**Comment by ChangelingRain** [comment]
Bombs are not good rewards, since they exist to explode shit.

**Comment by AnturK** [comment]
Maybe the cell rating is giving wrong impression, the initial idea was these are handed out when some threshold is reached (maxcap?) and were identical. The cargo multiplier is just there as placeholder.

**Comment by optimumtact** [comment]
I dont' think you understood joan, the act of creating the bomb should be the satisifying part, the bomb is just a side output

**Comment by AnturK** [comment]
Alright, i'll just close it until i have some actual use for this.

**Comment by ChangelingRain** [comment]
If you want to rework how bombs are made, oranges, then go for it.

**Comment by optimumtact** [comment]
Oh right I forgot I can't provide a point of view on literally anything unless I'm willing to do all the work in implementing it, sorry Joan, guess I missed that memo

**Comment by Nabski89** [comment]
Could it become a power cell instead of a cargo cell pointholder, helping out robotics?

Maybe depending on the size of your explosion it grants that level of tech research. Your standard maxcap because engineering 5? (or some other one).

Something productive and non destructive would be really nice to do with bombs.

**Pull request closed without merging**